### PR TITLE
Update contributing guidelines

### PR DESCRIPTION
**Pull request by vivek-koppuru**
*Issue #, if available:*

*Description of changes:*
Updated contributing guidelines to describe the processes this repository uses for planning and tracking issues as well as give more guidance on how to engage in discussions within this repository.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by TerryHowe** [comment]
I have a general concern if we should be having this in the docs, in this file CONTRIBUTING.md or both.  The easiest thing to do would be have this reference the doc, but is that cheating?

Also, I think Kirsten had a PR out to update this and many other files.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,10 +6,75 @@ documentation, we greatly value feedback and contributions from our community.
 Please read through this document before submitting any issues or pull requests to ensure we have all the necessary
 information to effectively respond to your bug report or contribution.
 
+## General Guidelines
+
+### Pull Requests
+
+Make sure to keep Pull Requests small and functional to make them easier to review and look up in commit history.
+* This repository uses "Squash and Commit" to keep our log clean and easy to find based on PR
```
Comment (TerryHowe) [inline CONTRIBUTING.md:14]:
```suggestion
* This repository uses "Squash and Commit" to keep our history clean and easy to find based on PR
```

Comment (TerryHowe) [reply CONTRIBUTING.md:14]: These don't really work as bullet items, maybe just make them sentences.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,10 +6,75 @@ documentation, we greatly value feedback and contributions from our community.
...
+* Easier to understand each PR functionally
+* Easier to revert changes based on PR
+
+Adding the appropriate docs/Changelog changes and unit/e2e tests as part of a feature is the responsibility of the
```
Comment (TerryHowe) [inline CONTRIBUTING.md:18]:
```suggestion
Adding the appropriate documentation, unit tests and e2e tests as part of a feature is the responsibility of the
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -6,10 +6,75 @@ documentation, we greatly value feedback and contributions from our community.
...
+feature owner, whether it is done in the same Pull Request or not.
+
+Pull Requests should follow the "subject: message" format, where the subject describes what part of the code is being
+modified.
+
+Refer to the [template](https://github.com/aws/eks-anywhere/blob/main/.github/PULL_REQUEST_TEMPLATE.md) for more information on what goes into a PR description.
+
+### Design Docs
+
+A contributor brings forward a design through a PR to allow for revisions and discussions on the repository.
```
Comment (TerryHowe) [inline CONTRIBUTING.md:28]:
```suggestion
A contributor proposes a design with a PR on the repository to allow for revisions and discussions.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -6,10 +6,75 @@ documentation, we greatly value feedback and contributions from our community.
...
+
+Categories:
+* Q/A - Questions
+* Proposals - Feature requests and other suggestions
+* Show and tell - Anything that the community would like to share
+* General - Everything else (possibly announcements as well)
+
+### GitHub Issues
+
+Used to file bugs, work items, and feature requests with actionable items/issues (Please refer to the "Reporting Bugs/Feature Requests" section below for more information).
```
Comment (TerryHowe) [inline CONTRIBUTING.md:45]:
```suggestion
GitHub issues are used to file bugs, work items, and feature requests with actionable items/issues (Please refer to the "Reporting Bugs/Feature Requests" section below for more information).
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -6,10 +6,75 @@ documentation, we greatly value feedback and contributions from our community.
...
+  * p1: don't have a set time but need to do
+  * p2: not currently being tracked (backlog)
+* "status/\<status\>" - status of the issue (notstarted, implementation, etc.)
+* "kind/\<kind\>" - type of issue (bug, feature, enhancement, etc.)
+
+Refer to the [template](https://github.com/aws/eks-anywhere/tree/main/.github/ISSUE_TEMPLATE) for more information on what goes into an issue description.
+
+### GitHub Milestones
+
+Used to plan work that is currently being tracked.
```
Comment (TerryHowe) [inline CONTRIBUTING.md:60]:
```suggestion
GitHub milestones are used to plan work that is currently being tracked.
```

**Review by TerryHowe — CHANGES_REQUESTED** [review]

**Comment by vivek-koppuru** [comment]
> I have a general concern if we should be having this in the docs, in this file CONTRIBUTING.md or both. The easiest thing to do would be have this reference the doc, but is that cheating?
> 
> Also, I think Kirsten had a PR out to update this and many other files.

I wanted to do that for Changelog to have it referenced in both the docs and here, let's see how that works out. I made these changes for now based on our meeting regarding GitHub and will make the updates accordingly.

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**